smartsim/experiment.py

```diff
@@ -276,6 +276,82 @@ def get_status(
         stats = (stats_map.get(i, InvalidJobStatus.NEVER_STARTED) for i in ids)
         return tuple(stats)
 
+    def wait(
+        self, *ids: LaunchedJobID, timeout: float | None = None, verbose: bool = True
+    ) -> None:
+        """Block execution until all of the provided launched jobs, represented
+        by an ID, have entered a terminal status.
+
+        :param ids: The ids of the launched ids to wait for.
+        :param timeout: The max time to wait for all of the launched jobs to end.
+        :param verbose: Whether found statuses should be displayed in the console.
+        :raises ValueError: No IDs were provided.
+        """
+        if not ids:
+            raise ValueError("No job ids to wait on provided")
+        self._poll_for_statuses(
+            ids,
```

Comment:
Mostly just because I didn't want to have to wait 5 seconds for each poll when I was testing the functionality of `_poll_for_statuses` in the unit tests (the 1,000 polls tests would then take > 1 hour hahaha).

It also allows for more flexible behavior, so if we use this method elsewhere, we do not need to strictly wait 5 seconds between polls, but that benefit is still largely theoretical at the moment. Its really just to speed up the tests for now.

Reply:
ah sounds bueno